ls1intum/Artemis
# `Iris`: Send feedback for each chat message

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
Like and dislike UI for each message for analytics purposes in the future

### Description
- Pressing like or dislike will dispatch an action to change `helpful` field in the message store
- Call API to save this field

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with Iris enabled

1. Log in to Artemis
2. Navigate to a Programming Exercise details page
3. Open chatbot
4. Chat something and rate the response
5. Cannot click on a rated message
6. Refresh the page and the rating should stay

#### Exam Mode Testing



### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage






| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
|  state-store.service.ts	 | 96% |        ✅                  |
| exercise-chat-widget.component.ts	 | 89.83% | ✅              |

### Screenshots



https://github.com/ls1intum/Artemis/assets/18320285/ca866e1c-fc4e-4f5c-a699-2c0ecf566e10
